src/components/select-field/Overlay.js

```diff
@@ -0,0 +1,116 @@
+// @flow
```

Comment:
Not sure if you already considered this, but Isn't this pretty similar to what Tether does for is in the other dropdown-style components? Could we use that for this intersection logic?

Reply:
Yeah, my initial implementation was to use the `Dropdown` component, but it turns out Tether will re-render each time the child changes causing a flicker in the dropdown which only really affects the `MultiSelectField` component where the dropdown remains open after selection in order to select other options.

Reply:
I had initially tried using `Dropdown` the component but found that Tether does a re-render when a child updates which causes a problem for the `MultiSelectField` because it remains open for multiple selections and causes a flicker

Reply:
I think he meant did you try using tether for this implementation, not drop-down. Like surrounding the ul with tether (which I am suggesting use popper instead)

Reply:
I didn't use Tether directly, but via `Dropdown`, but while debugging the flicker issue found that it's a limitation of Tether. I'll look into whether popper suffers from the same issue or not